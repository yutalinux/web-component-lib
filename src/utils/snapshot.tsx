import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
