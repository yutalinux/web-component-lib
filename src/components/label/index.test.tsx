import { snapshot } from "@/utils/tests/snapshot";
import { expect, test } from "vitest";
import Label from ".";

test("Label Snapshot", () => {
  expect(snapshot(Label, { label: "Label" })).toMatchSnapshot();
});
