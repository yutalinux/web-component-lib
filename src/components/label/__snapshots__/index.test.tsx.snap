// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label Snapshot 1`] = `"<label class="text-black dark:text-white">Label</label>"`;
